hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/compact/ReflectiveCompactSerializer.java

```diff
@@ -287,6 +261,14 @@ private void createFastReadWriteCaches(Class clazz) throws IOException {
                     }
                 };
                 writers[index] = (w, o) -> w.writeTimestampWithTimezone(name, (OffsetDateTime) field.get(o));
+            } else if (type.isEnum()) {
+                readers[index] = (reader, schema, o) -> {
+                    if (fieldExists(schema, name, UTF)) {
+                        String enumName = reader.readString(name);
+                        field.set(o, Enum.valueOf((Class<? extends Enum>) type, enumName));
+                    }
+                };
+                writers[index] = (w, o) -> w.writeString(name, ((Enum) field.get(o)).name());
```

Comment:
Similar comment as above. field.get(o) might return null. We could directly write null to writeString